### PR TITLE
Reorganized examples and added cleaner consumer and producer examples

### PR DESCRIPTION
**Pull request by edenhill**
Also removed the zookeeper example since it is of no use anymore.

### CODE REVIEW

File: examples/.gitignore [hunk 1]
```diff
@@ -1,9 +1,11 @@
 rdkafka_example
 rdkafka_performance
 rdkafka_example_cpp
-rdkafka_consumer_example
-rdkafka_consumer_example_cpp
+rdkafka_complex_consumer_example
```
Comment (rnpridgeon) [inline examples/.gitignore]: No ccloud example?

File: examples/producer.c [hunk 2]
```diff
@@ -225,7 +210,7 @@ int main (int argc, char **argv) {
                 } else {
                         fprintf(stderr, "%% Enqueued message (%zd bytes) "
                                 "for topic %s\n",
-                                len, rd_kafka_topic_name(rkt));
+                                len, topic);
                 }
```
Comment (rnpridgeon) [inline examples/producer.c]: You could also use the deliver report event to serve up your reports from the background queue no? We should provide an example of that.

Comment (edenhill) [reply examples/producer.c]: These are aimed to the be bare minimal examples, there's a bunch of existing more complex examples.

As for background thread delivery reports; another day :)

File: examples/.gitignore (hunk 1 above)
Comment (edenhill) [reply examples/.gitignore]: The [ccloud examples](https://github.com/confluentinc/examples/blob/5.3.1-post/clients/cloud/README.md) are linked from the top-level [README](https://github.com/edenhill/librdkafka#usage-in-code) thanks to @ybyzek

**Pull request merged**